### ISSUE TITLE
Installation deletes non-Tribler data within its folder

### ISSUE THREAD

**Issue by sofada** [Tribler/tribler#1659]
Hi developers,

I've installed Tribler into a custom folder (C:\...\Documents\Torrent) where I had placed also my downloads folder (...\Torrent\Downloads), and Tribler (the newest version 6.4.3) deleted all the content without any asking nor warning. The data (images, videos, music, .torrent files) even wasn't downloaded by Tribler.

What I did: Installed Tribler. Opened Tribler. Without setting up or turning up anything, I closed Tribler. Immediately I found out that my data within the Downloads folder disappeared (except two files containing diacritics in their name). 

I use Windows 8.1 64bit with all the latest updates.

**Comment by wkrick** [comment]
This raises several questions...
Why would someone install a program into their Documents folder?
Why would someone store their data in a folder inside the application installation folder?

**Comment by whirm** [comment]
Yeah, that's definitely a weird use case. That's an issue within the NSIS installer, and it left behind the non-ascii named files because it's unable to deal with them. 

We will see if we can add some sort of notice during the install procedure but there's not much more to do than that.

**Comment by BmetaMax** [comment]
I take back what I said on the forum, I did not do the same thing. My install folder and downloads folders are different.

**Comment by sofada** [comment]
> Why would someone install a program into their Documents folder?
> Why would someone store their data in a folder inside the application installation folder?

And why not? Your installer deletes user' data and you implicate that it is his fault because he used an unconventional folder location??

**Comment by BmetaMax** [comment]
@sofada Look, being defensive and churlish will get you no where. These guys are all volunteers. Sure it sucks, I'd be upset too- but in fairness it is a non-standard practice and you had to work to do that which means if it were me I'd be upset at myself as much as anyone.
Programs and data have been separated in most software for a long time, that's no coincidence- its a best practice to keep data safe from programs deleting what doesn't belong there (for them to run). 
Just take it on the chin, realize it was a mistake, and be nice when the volunteers try to help you! :)

**Comment by wkrick** [comment]
I'm not one of the developers.  I'm just another user asking questions about how common it would be for the bug to actually happen.

Though, I've seen this sort of thing before with overly aggressive uninstallers.  There was a video game in the late 90s that had a similar problem.  It was discovered when a user installed it to the root of their C: drive.  When they uninstalled it, it tried to delete everything on his drive.

**Comment by whirm** [comment]
I _am_ a Tribler dev. :)

This bug has been there for I would say ~7 years (As long as there's been an installer for Tribler) and nobody complained about it before. As you can see it's rather uncommon to do what you did.

We are going to have a look and see if we can add a warning or force the user to choose a nonexisting dir to mitigate this. But if you copy stuff into that dir and then uninstall Tribler, the data will be gone anyway.

We are using the NSIS stuff to generate the installer and it's quite old and crufty (Try running the installer from a non ASCII path) but we can't invest time in migrating to something more up to date right now (we gladly accept contributions though).
